src/integration-test/java/com/commercetools/sync/integration/types/TypeSyncIT.java

```diff
@@ -0,0 +1,837 @@
+
+        verify(spyTypeSyncOptions)
+                .applyErrorCallback(eq("Failed to create type of key 'key_2'."), any());
+
+        AssertionsForStatistics.assertThat(typeSyncStatistics).hasValues(1, 0, 0, 1);
+    }
+
+    @Test
+    public void sync_WithErrorUpdatingTheTypeInCT_ShouldExecuteCallbackOnErrorAndIncreaseFailedCounter() {
+        final TypeDraft newTypeDraft = TypeDraftBuilder.of(
+                TYPE_KEY_1,
+                TYPE_NAME_1,
+                ResourceTypeIdsSetBuilder.of().addCategories().build())
+                .description(TYPE_DESCRIPTION_1)
+                .fieldDefinitions(singletonList(FIELD_DEFINITION_1))
+                .build();
+
+        final TypeSyncOptions typeSyncOptions = TypeSyncOptionsBuilder
+                .of(CTP_TARGET_CLIENT)
+                .build();
+
+        TypeSyncOptions spyTypeSyncOptions = spy(typeSyncOptions);
+
+        TypeService typeService = new TypeServiceImpl(spyTypeSyncOptions);
+        TypeService spyTypeService = spy(typeService);
+
+        doReturn(supplyAsync(() -> {
+            throw new SphereException();
+        })).when(spyTypeService).updateType(Mockito.any(), Mockito.anyList());
+
+        final TypeSync typeSync = new TypeSync(spyTypeSyncOptions, spyTypeService);
+
+        final TypeSyncStatistics typeSyncStatistics = typeSync
+                .sync(singletonList(newTypeDraft))
+                .toCompletableFuture().join();
+
+        verify(spyTypeSyncOptions)
+                .applyErrorCallback(eq("Failed to update type of key 'key_1'."), any());
+
+        AssertionsForStatistics.assertThat(typeSyncStatistics).hasValues(1, 0, 0, 1);
+    }
+
+    @Test
+    public void sync_WithoutName_ShouldExecuteCallbackOnErrorAndIncreaseFailedCounter() {
+        final TypeDraft newTypeDraft = TypeDraftBuilder.of(
+                TYPE_KEY_1,
+                null,
+                ResourceTypeIdsSetBuilder.of().addCategories().build())
+                .description(TYPE_DESCRIPTION_1)
+                .fieldDefinitions(asList(FIELD_DEFINITION_1, FIELD_DEFINITION_2))
+                .build();
+
+        final TypeSyncOptions typeSyncOptions = TypeSyncOptionsBuilder
+                .of(CTP_TARGET_CLIENT)
+                .build();
+
+        TypeSyncOptions spyTypeSyncOptions = spy(typeSyncOptions);
+
+        final TypeSync typeSync = new TypeSync(spyTypeSyncOptions);
+
+        final TypeSyncStatistics typeSyncStatistics = typeSync
+                .sync(singletonList(newTypeDraft))
+                .toCompletableFuture().join();
+
+        // Since the error message and exception is coming from commercetools, we don't test the actual message and
+        // exception
+        verify(spyTypeSyncOptions).applyErrorCallback(any(), any());
+
+        AssertionsForStatistics.assertThat(typeSyncStatistics).hasValues(1, 0, 0, 1);
+    }
+
+    @Test
+    public void sync_WithoutFieldDefinitionType_ShouldExecuteCallbackOnErrorAndIncreaseFailedCounter() {
```

Comment:
Yes, it will throw, For the API it's a validation 400 error because the name is a required value. In this case, it just asserts the failed counter. If you want to get a throwable you should add an error callback.